lighthouse-core/audits/font-display.js

```diff
@@ -33,59 +37,101 @@ class FontDisplay extends Audit {
       title: str_(UIStrings.title),
       failureTitle: str_(UIStrings.failureTitle),
       description: str_(UIStrings.description),
-      requiredArtifacts: ['devtoolsLogs', 'Fonts'],
+      requiredArtifacts: ['devtoolsLogs', 'CSSUsage', 'URL'],
     };
   }
 
+  /**
+   * @param {LH.Artifacts} artifacts
+   */
+  static findPassingFontDisplayDeclarations(artifacts) {
+    /** @type {Set<string>} */
+    const passingURLs = new Set();
+
+    // Go through all the stylesheets to find all @font-face declarations
+    for (const stylesheet of artifacts.CSSUsage.stylesheets) {
+      // Eliminate newlines so we can more easily scan through with a regex
+      const newlinesStripped = stylesheet.content.replace(/\n/g, ' ');
+      // Find the @font-faces
+      const fontFaceDeclarations = newlinesStripped.match(/@font-face\s*{(.*?)}/g) || [];
+      // Go through all the @font-face declarations to find a declared `font-display: ` property
+      for (const declaration of fontFaceDeclarations) {
+        const rawFontDisplay = declaration.match(/font-display:(.*?);/);
+        // If they didn't have a font-display property, it's the default, and it's failing; bail
+        if (!rawFontDisplay) continue;
+        // If they don't have one of the passing font-display values, it's failing; bail
+        const hasPassingFontDisplay = PASSING_FONT_DISPLAY_REGEX.test(rawFontDisplay[0]);
+        if (!hasPassingFontDisplay) continue;
+
+        // If it's passing, we'll try to find the URL it's referencing.
+        const rawFontURLs = declaration.match(CSS_URL_GLOBAL_REGEX);
+        // If no URLs, we can't really do anything; bail
+        if (!rawFontURLs) continue;
+
+        const relativeURLs = rawFontURLs
+          // @ts-ignore - guaranteed to match from previous regex, pull URL group out
+          .map(s => s.match(CSS_URL_REGEX)[1].trim())
+          .map(s => {
+            // remove any quotes surrounding the URL
+            if (/^('|").*\1$/.test(s)) {
+              return s.substr(1, s.length - 2);
+            }
+
+            return s;
+          });
+
+        // Convert the relative CSS URL to an absolute URL and add it to the passing set
+        for (const relativeURL of relativeURLs) {
+          try {
+            const absoluteURL = new URL(relativeURL, artifacts.URL.finalUrl);
+            passingURLs.add(absoluteURL.href);
+          } catch (err) {
+            Sentry.captureException(err, {tags: {audit: this.meta.id}});
```

Comment:
We should probably only capture once per audit run. Kind of annoying to implement, but some sites end up with a ridiculous number of broken URLs (or whatever we're capturing in a loop). See #5994 (which the only one I had actually checked off was the gatherer you're deleting :)...or disagree with the conclusion there :)

Reply:
I have a counter-proposal that I will put up in a different PR :)